jrugged-core/src/main/java/org/fishwife/jrugged/SkepticBreaker.java

```diff
@@ -0,0 +1,471 @@
+        else {
+            r.run();
+        }
+    }
+
+    /** Wrap the given service call with the {@link SkepticBreaker}
+     *  protection logic.
+     *  @param r the {@link Runnable} to attempt
+     *  @param result what to return after <code>r</code> succeeds
+     *  @return result
+     *  @throws BreakerException if the
+     *    breaker was OPEN and this attempt wasn't the
+     *    reset attempt
+     *  @throws Exception if <code>c</code> throws one during
+     *    execution
+     */
+    public <V> V invoke(Runnable r, V result) throws Exception {
+        if (!byPass) {
+            if (!allowRequest()) {
+            	if(!isHardTrip){
+            		try{
+            			Runnable mockRunnable = createMock(Runnable.class);
+            		    mockRunnable.run(); 
+            		}
+            		catch (Throwable cause){
+            			handleFailure(cause); 
+            		}
+            	}
+                throw mapException(new BreakerException());
+            }
+
+            try {
+                r.run();
+                close();
+                return result;
+            } catch (Throwable cause) {
+                handleFailure(cause);
+            }
+            throw new IllegalStateException("not possible");
+        }
+        else {
+            r.run();
+            return result;
+        }
+    }
+
+    /**
+     * Causes the {@link SkepticBreaker} to trip and OPEN; no new
+     *  requests will be allowed until the <code>SkepticBreaker</code>
+     *  resets. Skeptic level increases and timers are updated accordingly.
+     */
+    public void trip() {
+    	if (state != BreakerState.OPEN) {
+            openCount.getAndIncrement();
+        }
+    	if (skepticLevel.get() < maxLevel.get()) {
+    		increaseSkepticLevel();
+    	}
+        state = BreakerState.OPEN;
+        lastFailure.set(System.currentTimeMillis());
+
+        notifyBreakerStateChange(getStatus());
+    }
+    
+    /**
+     * Increments the skeptic level and updates timers; occurs when the 
+     * {@link SkepticBreaker} trips and moves from CLOSED to OPEN
+     */
+    public void increaseSkepticLevel() {
+    	skepticLevel.set(skepticLevel.incrementAndGet());
+    	updateTimers();
+    }
+    
+    /**
+     * Updates wait timer and good timer based on current skeptic level.
+     */
+    public void updateTimers() {
+    	waitTime.set((long) (waitBase.get() + waitMult.get() * Math.pow(2, skepticLevel.get())));
+    	goodTime.set((long) (goodBase.get() + goodMult.get() * Math.pow(2, skepticLevel.get())));
+    }
+
+    /**
+     * Manually set the breaker to be reset and ready for use.  This
+     * is only useful after a manual trip otherwise the breaker will
+     * trip automatically again if the service is still unavailable.
+     * Just like a real breaker.  WOOT!!!
+     */
+    public void reset() {
+        state = BreakerState.CLOSED;
+        isHardTrip = false;
+        byPass = false;
+        skepticLevel.set(0);
```

Comment:
Silly - but I would likely have put this in a method called resetSkepticLevel() so that it told us what it was being done for.